Plist2swift/Plist2swift.swift

```diff
@@ -83,6 +83,69 @@ private func generateHeader() {
 		""")
 }
 
+/**
+Checking the key is present in the Optional Dictionary
+*/
+private func isKeyPresentInOptionalDictionary(keyToSearch: String, tupleKey: String, optionalDictionary: [String: [String: String]]) -> Bool {
+	guard let optionalKeysAndTypes = optionalDictionary[keyToSearch] else {
+		return false
+	}
+
+	let optionalArray = optionalKeysAndTypes.keys
+	return optionalArray.contains(tupleKey)
+}
+
+/**
+Checking the key is present in  all the plists
+*/
+private func isKeyAvailableInAllPlists(keyToSearch: String, tupleKey: String, tuplesForPlists: [String: KeyValueTuples]) -> Bool {
+	for plistPath in tuplesForPlists.keys {
+
+		guard
+			let tuples = tuplesForPlists[plistPath],
+			let dictionary = tuples[tupleKey] as? Dictionary<String, Any> else {
+			return false
+		}
+
+		if (dictionary.keys.contains(keyToSearch) == false) {
+			return false
+		}
+	}
+
+	return true
+}
+
+/**
+Generate Protocol for the Tuples and return the optional dictionary
+*/
+private func generateProtocol(tuplesForPlists: [String: KeyValueTuples], allKeyValueTuples: [String: KeyValueTuples]) -> [String: [String: String]] {
+	var optionalDictionary = [String: [String: String]]()
```

Comment:
I would prefer `optionalsDictionary`, read as “dictionary with optional values”, because optional dictionary sounds like a generic dictionary that is optional, as in `?`.